s3_upload/utils/utils.py

```diff
@@ -210,6 +216,75 @@ def read_config(config) -> dict:
         return json.load(fh)
 
 
+def verify_args(args) -> None:
+    """
+    Verify that the provided args are valid
+
+    Parameters
+    ----------
+    args : argparse.NameSpace
+        parsed command line arguments
+    """
+    # TODO - complete this once I decide on all args to have
+    pass
+
+
+def verify_config(config) -> None:
+    """
+    Verify that config structure and parameters are valid
+
+    Parameters
+    ----------
+    config : dict
+        contents of config file to check
+    """
+    log.debug(
+        "Verifying contents of config are valid, contents parsed: %s", config
+    )
+    errors = []
+
+    if not isinstance(config.get("max_cores", 0), int):
+        errors.append("max_cores must be an integer")
+
+    if not isinstance(config.get("max_threads", 0), int):
+        errors.append("max_threads must be an integer")
+
+    if not config.get("log_dir"):
+        errors.append("required parameter log_dir not defined")
+
+    if not config.get("monitor"):
+        errors.append("required parameter monitor not defined")
+
+    for idx, monitor in enumerate(config.get("monitor", "")):
+        for key, expected_type in {
+            "monitored_directories": list,
+            "bucket": str,
+            "remote_path": str,
+        }.items():
+            if not monitor.get(key):
+                errors.append(
+                    f"required parameter {key} missing from monitor section"
+                    f" {idx}"
+                )
+            else:
+                if not isinstance(monitor.get(key), expected_type):
+                    errors.append(
+                        f"{key} not of expected type from monitor section "
+                        f"{idx}. Expected: {expected_type} | Found "
+                        f"{type(monitor.get(key))}"
+                    )
+
+    if errors:
+        error_message = (
+            f"{len(errors)} errors found in config:{chr(10)}{chr(9)}"
```

Comment:
What does the chr(10) and chr(9) mean? I'm not familiar with it and google says it represents a specified unicode, but I can't seem to find what these mean exactly. Thanks!